### PR TITLE
[133886] Refactoring: Change scheme to specify supervisor spec

### PR DESCRIPTION
**Pull request by ttokit**
https://acsmine.tok.access-company.com/redmine/issues/133886
* Changed to use ["Streamlined child specs"](https://elixir-lang.org/blog/2017/07/25/elixir-v1-5-0-released/#streamlined-child-specs) bacause `Supervisor.Spec` is deprecated.

### CODE REVIEW

File: core/sup_tree_core/cluster_hosts_poller.ex [hunk 1]
```diff
@@ -21,7 +21,7 @@ defmodule AntikytheraCore.ClusterHostsPoller do
     end
   end
 
-  def start_link() do
+  def start_link(_args) do
```
Comment (skirino) [inline core/sup_tree_core/cluster_hosts_poller.ex]: Let's change `_args` to `[]` to make the incoming arguments explict.

File: lib/gear_application/gear_application.ex [hunk 2]
```diff
@@ -29,7 +28,7 @@ defmodule Antikythera.GearApplication do
   alias AntikytheraCore.GearLog.Writer
 
   @doc false
-  defun start(gear_name :: v[GearName.t], children :: [Spec.spec]) :: {:ok, pid} do
+  defun start(gear_name :: v[GearName.t], children :: list) :: {:ok, pid} do
```
Comment (ymtszw) [inline lib/gear_application/gear_application.ex]: Typespec here can be improved.
This `children` goes directly into [Supervisor.start_link/2](https://hexdocs.pm/elixir/1.5.3/Supervisor.html#start_link/2), in that the spec for `children` is `[:supervisor.child_spec | {module, term} | module]`.
(Notice the above link is for Elixir 1.5.3)

It is a bit verbose so let us put:
```ex
@type child_spec :: [:supervisor.child_spec | {module, term} | module]
```
and use it from other places including updated gear_template.

File: core/sup_tree_core/cluster_hosts_poller.ex (hunk 1 above)
Comment (ttokit) [reply core/sup_tree_core/cluster_hosts_poller.ex]: fixed in a2bb8aa

File: lib/gear_application/gear_application.ex (hunk 2 above)
Comment (ttokit) [reply lib/gear_application/gear_application.ex]: fixed in 44fa02c

File: lib/gear_application/gear_application.ex [hunk 3]
```diff
@@ -28,8 +27,10 @@ defmodule Antikythera.GearApplication do
   alias AntikytheraCore.Config.Gear, as: GearConfig
   alias AntikytheraCore.GearLog.Writer
 
+  @type child_spec :: [:supervisor.child_spec | {module, term} | module]
```
Comment (skirino) [inline lib/gear_application/gear_application.ex]: This should represent spec of 1 child.
```ex
@type child_spec :: :supervisor.child_spec | {module, term} | module
```
Then use this for parameter types as `children :: [child_spec]`.
We may also define `@type child_specs :: [child_spec]` but we probably don't need it.

Comment (ttokit) [reply lib/gear_application/gear_application.ex]: fixed in c9f159e

**Comment by skirino** [comment]
LGTM!

File: core/sup_tree_core/executor_pool/executor_pool.ex [hunk 4]
```diff
@@ -129,9 +137,17 @@ defmodule AntikytheraCore.ExecutorPool do
...
+    def child_spec(args) do
+      %{
+        id:       __MODULE__,
+        start:    {__MODULE__, :start_link, [args]},
+        shutdown: :infinity,
+        type:     :supervisor,
+      }
+    end
+
+    defun start_link(_arg) :: {:ok, pid} do
```
[the remark this answers is not in the document]
Comment (ttokit) [reply core/sup_tree_core/executor_pool/executor_pool.ex]: Oops, I'd overlooked.
fixed in ec6fa95

**Comment by ymtszw** [comment]
# okay 🚀

**Pull request merged**